andrepereiradasilva/joomla-cms
# JLanguage(Helper) improvements

### Summary of Changes

This PR does several things, all language related:
- adds a new `JLanguageHelper::getInstalledLanguages()` to get the installed language from DB. This method uses static var so the query is only run once.
- Moves the following method from `JLanguage` to `JLanguageHelper` and deprecate the old ones. Since those method does not needs to be in `JLanguage` as this methods doesn't do any interactions with `JLanguage` class neither depends on it.

```
JLanguage::exists() -> JLanguageHelper::exists()
JLanguage::getMetadata() -> JLanguageHelper::getMetadata()
JLanguage::getKnownLanguages() -> JLanguageHelper::getKnownLanguages()
JLanguage::getLanguagePath() -> JLanguageHelper::getLanguagePath()
JLanguage::parseLanguageFiles() -> JLanguageHelper::parseLanguageFiles()
JLanguage::parseXMLLanguageFile() -> JLanguageHelper::parseXMLLanguageFile()
JLanguageMultilang::getSiteLangs() -> JLanguageHelper::getInstalledLanguages(0)
```
- Does the necessary changes in all cores to not use now deprecated methods.
- Makes the necessary changes across the core for the language extension db queries to use the new `JLanguageHelper::getInstalledLanguages()`.
- Does some minor code improvements and micro optimizations in this methods.

Note: Most of this changes were already done by the joomla framework language package (see https://github.com/joomla-framework/language/blob/2.0-dev/src/LanguageHelper.php).
### Testing Instructions
1. Code review
2. Use latest staging multilingual site
3. Apply patch and confirm all language related things are working properly
   - Confirm installed language lists work fine (user profile site/admin language, backend login list)
   - Confirm installed languages view in com_languages shows all installed langauges.
   - Confirm language switch and language filter plugin works fine
   - Confirm install language works fine
4. Confirm installation language selector and install languages works fine
### Documentation Changes Required

None.
